src/realm/util/file_mapper.cpp

```diff
@@ -241,13 +241,8 @@ static dispatch_queue_t reclaimer_queue;
 static void ensure_reclaimer_thread_runs()
 {
     if (!reclaimer_timer) {
-        if (__builtin_available(iOS 10, macOS 12, tvOS 10, watchOS 3, *)) {
```

Comment:
Both of these removed checks for macOS 12 were incorrect and should have been 10.12.